modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -1065,24 +1080,46 @@ public void cancelLocalSnapshotTask(String name) {
 
         GridKernalContext kctx0 = cctx.kernalContext();
 
-        Collection<ClusterNode> bltNodes = F.view(cctx.discovery().serverNodes(AffinityTopologyVersion.NONE),
-            (node) -> CU.baselineNode(node, kctx0.state().clusterState()));
-
-        kctx0.task().setThreadContext(TC_SKIP_AUTH, true);
-        kctx0.task().setThreadContext(TC_SUBGRID, bltNodes);
-
         kctx0.task().execute(SnapshotMetadataCollectorTask.class, name).listen(f0 -> {
             if (f0.error() == null) {
                 Map<ClusterNode, List<SnapshotMetadata>> metas = f0.result();
 
                 Map<Integer, String> grpIds = grps == null ? Collections.emptyMap() :
                     grps.stream().collect(Collectors.toMap(CU::cacheId, v -> v));
 
+                byte[] currentMasterKeyDigest = kctx0.config().getEncryptionSpi().masterKeyDigest();
+
                 for (List<SnapshotMetadata> nodeMetas : metas.values()) {
-                    for (SnapshotMetadata meta : nodeMetas)
+                    for (SnapshotMetadata meta : nodeMetas) {
+                        if (meta.masterKeyDigest() == null)
+                            continue;
```

Comment:
Should we remove the meta partitions from the `grpIds` collection (see 1113 line)? Now your changes affect only encrypted cached, what about non-encrypted?

Reply:
What is "meta partitions"? it is snapshot metadata. We check encryption, master keys here. However, another fixes applied.

Reply:
Fixed